src/components/pages/parameter-or-variable.jsx

```diff
@@ -20,15 +21,39 @@ const ParameterOrVariablePage = React.createClass({
     currency: PropTypes.string.isRequired,
     location: locationShape.isRequired,
     params: PropTypes.shape({ name: PropTypes.string.isRequired }).isRequired, // URL params
-    parameters: PropTypes.arrayOf(AppPropTypes.parameterOrScale).isRequired,
+    parameters: PropTypes.objectOf(AppPropTypes.parameterOrScale).isRequired,
     variables: PropTypes.arrayOf(AppPropTypes.variable).isRequired,
   },
+  getInitialState() {
+    const variable = this.props.variables.find(variable => variable.name === this.props.params.name)
+    return {variable: variable, waitingForResponse: true}
```

Comment:
It's not related to React: I'd say it's about following a principle of least surprise, minimizing the number of places I have to look at when I want to know what properties an may object have. But don't change anything, indeed in JS it's common to add properties after the object is created.